lib/pure/collections/tableimpl.nim

```diff
@@ -101,29 +96,16 @@ template delImpl() {.dirty.} =
 
 template clearImpl() {.dirty.} =
   for i in 0 ..< t.dataLen:
-    when compiles(t.data[i].hcode): # CountTable records don't contain a hcode
+    when t.data is KeyValuePairSeq: # CountTable records don't contain a hcode
       t.data[i].hcode = 0
     t.data[i].key = default(type(t.data[i].key))
     t.data[i].val = default(type(t.data[i].val))
   t.counter = 0
 
-template ctAnd(a, b): bool =
-  # pending https://github.com/nim-lang/Nim/issues/13502
-  when a:
-    when b: true
-    else: false
-  else: false
-
 template initImpl(result: typed, size: int) =
-  when ctAnd(declared(SharedTable), type(result) is SharedTable):
```

Comment:
``SharedTable`` is currently not well tested (if at all) but people use it so don't touch this code.

Reply:
Then tell me what is not well tested about it, and I will improve upon it.

Reply:
> SharedTable is currently not well tested (if at all) but people use it so don't touch this code.

indeed, you're introducing another regression, see https://github.com/nim-lang/Nim/pull/13687#issuecomment-602307527
such modifications should not be bundled in a giant PR with tons of unrelated changes